### PR TITLE
Null check handling for Stack trace ambiguity #543

### PR DESCRIPTION
**Pull request by SougandhS**
## What it does

Fix for #543 
## Author checklist

- [x] I have thoroughly tested my changes
- [x] The change is following the [coding conventions](https://wiki.eclipse.org/Platform/How_to_Contribute#Coding_Conventions)
- [x] I have signed the [Eclipse Contributor Agreement (ECA)](https://www.eclipse.org/legal/ECA.php)

### CODE REVIEW

**Comment by iloveeclipse** [comment]
Looking on the number of issues I've found above, I will now revert https://github.com/eclipse-jdt/eclipse.jdt.debug/pull/498 PR as 
1) the code has multiple regressions
2) The code in  `org.eclipse.jdt.internal.debug.ui.console.JavaStackTraceHyperlink.searchCompleted(Object, String, int, IStatus)` is simply not maintainable.

I would recommend to rewrite https://github.com/eclipse-jdt/eclipse.jdt.debug/pull/498.

File: org.eclipse.jdt.debug.ui/ui/org/eclipse/jdt/internal/debug/ui/console/JavaStackTraceHyperlink.java [hunk 1]
```diff
@@ -208,6 +208,33 @@ public IStatus runInUIThread(IProgressMonitor monitor) {
 					List<Object> matches = (List<Object>) source;
 					List<Object> exactMatchesFiltered = new ArrayList<>();
 					String originalHyperLink2 = originalHyperLink;
+					if (originalHyperLink2 == null) {
```
[the remark this answers is not in the document]
Comment (SougandhS) [reply org.eclipse.jdt.debug.ui/ui/org/eclipse/jdt/internal/debug/ui/console/JavaStackTraceHyperlink.java:211]: Sure @iloveeclipse 👍

**Comment by SougandhS** [comment]
> Looking on the number of issues I've found above, I will now revert #498 PR as
> 
> 1. the code has multiple regressions
> 2. The code in  `org.eclipse.jdt.internal.debug.ui.console.JavaStackTraceHyperlink.searchCompleted(Object, String, int, IStatus)` is simply not maintainable.
> 
> I would recommend to rewrite #498.

Will re-do this like you suggested.

**Comment by SougandhS** [comment]
> 1. Commit message is wrong and doesn't explain the fix, see https://github.com/eclipse-jdt/.github/blob/main/CONTRIBUTING.md#commit-message-recommendations
> 2. With this fix, instead of an error I get this dialog:
>    image
>    and of course this makes no sense, **I don't want to create any brekpoints**!
>    This was **before** [using method name to find unqualified Stacktraces in Java Stack Trace #498](https://github.com/eclipse-jdt/eclipse.jdt.debug/pull/498):
>    image
> 3. Clocking on `ForkJoinTask` in `at java.base/java.util.concurrent.ForkJoinTask.reportExecutionException(ForkJoinTask.java:605)` shows empty and meaningless error about opening something from clipboard (???):
> 
> image
> 
> while it was type selection dialog before : image

Hi @iloveeclipse 

Regarding 2. 

It seems to be the default behaviour -> In master I get the same behaviour now too. If the generated source is not list and have only 1 binaryType and is of exception type then this implementation is called -> 

org.eclipse.jdt.internal.debug.ui.console.JavaExceptionHyperLink.processSearchResult() instead of org.eclipse.jdt.internal.debug.ui.console.JavaStackTraceHyperlink.processSearchResult() 

I reproduced the same now.
If I have only 1 BinaryType then control goes here 
image
then -> 
image


But If I have multiple BinaryType then openType is triggered 
image

Shall I raise this as a new issue ?

**Comment by iloveeclipse** [comment]
> Shall I raise this as a new issue ?

I don't have IDE right now, but any bug you see in a plain SDK without any changes is a bug and should be reported. If you see "add breakpoint" dalog for clicking on "regular" stack trace line, instead of navigation to the right code, it is a bug.

**Comment by SougandhS** [comment]
> > Shall I raise this as a new issue ?
> 
> I don't have IDE right now, but any bug you see in a plain SDK without any changes is a bug and should be reported. If you see "add breakpoint" dalog for clicking on "regular" stack trace line, instead of navigation to the right code, it is a bug.

on regular stacktrace it is not going to any breakpoint creation.. its only for Exception traces . I believe its a feature, but like you reported I dont feel its a good approach. can we provide some right click option for creating breakpoint option rather than directly showing this ?

**Pull request closed without merging**